src/coreclr/tools/r2rdump/CoreDisTools.cs

```diff
@@ -1392,6 +1392,212 @@ private void AnalyzeRiscV64Itype(uint instruction, out uint rd, out uint rs1, ou
+            {
+                offs = (int)(((ins >> 10) & 0xffff) | ((ins & 0x3ff) << 16)) << 6;
+                offs >>= 4;
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Determine whether a given instruction is a BZ(BEQZ, BNEZ, BCEQZ, BCNEZ).
+        /// </summary>
+        /// <param name="ins">Assembly code of instruction</param>
+        private bool IsLoongArch64BZInstruction(uint ins, out int offs)
+        {
+            uint Opcode = (ins >> 26) & 0x3f;
+            offs = 0;
+            if ((Opcode == 0x10) || (Opcode == 0x11) || (Opcode == 0x12))
+            {
+                offs = (int)((((ins >> 10) & 0xffff) | ((ins & 0x1f) << 16)) << 11);
+                offs >>= 9;
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Determine whether a given instruction is a JIRL RA.
+        /// </summary>
+        /// <param name="ins">Assembly code of instruction</param>
+        private bool IsLoongArch64JirlRAInstruction(uint ins, out uint rj, out int offs)
+        {
+            rj   = 0;
+            offs = 0;
+            if ((((ins >> 26) & 0x3f) == 0x13) && ((ins & 0x1f) == 1))
+            {
+                rj = (ins >> 5) & 0x1f;
+                offs = (short)((ins >> 10) & 0xffff);
+                offs <<= 2;
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Determine whether a given instruction is a PCADDU12I.
+        /// </summary>
+        /// <param name="ins">Assembly code of instruction</param>
+        private bool IsLoongArch64Pcaddu12iInstruction(uint ins, out uint rd, out int imm)
+        {
+            rd = 0;
+            imm = 0;
+            if (((ins >> 25) & 0x3f) == 0xe)
+            {
+                rd = ins & 0x1f;
+                imm = (int)((ins >> 5) & 0xfffff) << 12;
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Determine whether a given instruction is a LD.D or ADDI.D.
+        /// </summary>
+        /// <param name="ins">Assembly code of instruction</param>
+        private bool IsLoongArch64Ld_dOrAddi_dInstruction(uint ins, out uint rd, out uint rj, out int imm)
+        {
+            imm = 0;
+            rd = rj = 0;
+
+            if ((((ins >> 22) & 0x3ff) == 0xa3) || (((ins >> 22) & 0x3ff) == 0xb))
+            {
+                rd = ins & 0x1f;
+                rj = (ins >> 5) & 0x1f;
+                imm = (int)((ins >> 10) & 0xfff) << 20;
+                imm >>= 20;
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Determine whether a given instruction is a FLD.D.
+        /// </summary>
+        /// <param name="ins">Assembly code of instruction</param>
+        private bool IsLoongArch64FLD_DInstruction(uint ins)
```

Comment:
This uses `FLD_D`. The earlier method uses `Ld_d`. Could you please make the casing more consistent between different methods?

Reply:
Ok, thanks.